File: quiz-platform-backend/src/main/java/com/quizplatform/service/QuestionService.java
package com.quizplatform.service;

import com.quizplatform.dto.request.CreateQuestionRequest;
import com.quizplatform.dto.request.UpdateQuestionRequest;
import com.quizplatform.dto.response.QuestionResponse;

import java.util.List;

public interface QuestionService {
    QuestionResponse addQuestion(Long quizId, CreateQuestionRequest request);
    QuestionResponse updateQuestion(Long questionId, UpdateQuestionRequest request);
    void deleteQuestion(Long questionId);
    List<QuestionResponse> getQuestionsByQuizId(Long quizId, boolean includeAnswers);
    QuestionResponse getQuestionById(Long questionId, boolean includeAnswers);
    long countByQuizId(Long quizId);
}

File: quiz-platform-frontend/src/services/questionService.ts
import api from '@/utils/api';
import {
  ApiResponse,
  CreateQuestionRequest,
  QuestionResponse,
} from '@/types';

const questionService = {
  getQuestions: async (quizId: number) => {
    const response = await api.get<ApiResponse<QuestionResponse[]>>(
      `/quizzes/${quizId}/questions`
    );

    return response.data;
  },

  addQuestion: async (
    quizId: number,
    request: CreateQuestionRequest
  ) => {
    const response = await api.post<ApiResponse<QuestionResponse>>(
      `/quizzes/${quizId}/questions`,
      request
    );

    return response.data;
  },

  updateQuestion: async (
    questionId: number,
    request: CreateQuestionRequest
  ) => {
    const response = await api.put<ApiResponse<QuestionResponse>>(
      `/questions/${questionId}`,
      request
    );

    return response.data;
  },

  deleteQuestion: async (questionId: number) => {
    const response = await api.delete<ApiResponse<void>>(
      `/questions/${questionId}`
    );

    return response.data;
  },
};

export default questionService;